### PR TITLE
timestamps verification: verify against relay slots

### LINKED ISSUE #2219: Bug Fix for `pallet_timestamp::Config`
Details recorded here: https://docs.google.com/document/d/1MfF35Ry9Oni9Y3WxvZbwJIVVoYuiXu2E7Nuq7BhAf_k/edit?usp=sharing

### PR DESCRIPTION
**Pull request by saraswatpuneet**
Verify  timestamps against relay chain slot numbers via `OnTimestampSet`.

Closes  #2219 

# Discussion

Solution proposed in story above

# Checklist
- [ ] Updated Pallet Readme?
- [ ] Updated js/api-augment for Custom RPC APIs?
- [ ] Design doc(s) updated?
- [ ] Unit Tests added?
- [ ] e2e Tests added?
- [ ] Benchmarks added?
- [x] Spec version incremented?

### CODE REVIEW

**Review by enddynayn — APPROVED** [review]
🚀

File: runtime/frequency/src/lib.rs [hunk 1]
```diff
@@ -655,6 +655,9 @@ impl pallet_time_release::Config for Runtime {
 impl pallet_timestamp::Config for Runtime {
 	/// A timestamp: milliseconds since the unix epoch.
 	type Moment = u64;
+	#[cfg(any(not(feature = "frequency-no-relay"), feature = "frequency-lint-check"))]
+	type OnTimestampSet = Aura;
+	#[cfg(feature = "frequency-no-relay")]
```
Comment (saraswatpuneet) [inline runtime/frequency/src/lib.rs:660]: Aura is not available in no-relay mode

File: runtime/frequency/src/lib.rs [hunk 2]
```diff
@@ -655,6 +655,9 @@ impl pallet_time_release::Config for Runtime {
 impl pallet_timestamp::Config for Runtime {
 	/// A timestamp: milliseconds since the unix epoch.
 	type Moment = u64;
+	#[cfg(any(not(feature = "frequency-no-relay"), feature = "frequency-lint-check"))]
```
Comment (aramikm) [inline runtime/frequency/src/lib.rs:658]: why do we lint check on this one?

Comment (saraswatpuneet) [reply runtime/frequency/src/lib.rs]: right, wasnt needed for this, fixed it

**Review by wilwade — APPROVED** [review]
Looks good to me!

**Review by enddynayn — APPROVED** [review]
👍 great!

**Review by mattheworris — APPROVED** [review]
🚢 it

**Pull request merged**